api/v1/iprange_types.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2024 Swisscom (Schweiz) AG.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="self.startAddress < self.endAddress"
```

This is tested and worked in my local netbox. @alexandernorth @bruelea what do you think?

Reply:
After discussion, we agree that for IPv6, it might not work due to the various formats you can represent it, so we will NOT implement this validation now